### ISSUE TITLE
Application access returning "header size too large" starting in v11

### ISSUE THREAD

**Issue by rcanderson23** [gravitational/teleport#21731]
Expected behavior:
Application access allowed access to internal application with agent version v10. Access should continue without error after upgrading teleport app agent to v11.

Current behavior:
Application access works in v10.3.1. After upgrading to v11.2.1, application returns an error "Header size too large".

Bug details:
- Teleport version
  - Cluster is v11.3.3
  - Working in v10.3.1. Error observed in v11.2.1, v11.2.2, v11.2.3, v11.3.2.
- Debug logs
```
2023-02-13T16:54:29Z INFO [AUDIT]     app.session.chunk app_name:sales-center-dev app_public_addr:redacted.platform.teleport.sh app_uri:https://redacted.cluster.internal cluster_name:platform.teleport.sh code:T2008I ei:0 event:app.session.chunk namespace:default server_id:fc7dd5dd-7227-43e2-b651-613a2dd09396 session_chunk_id:bc6c10e4-1c29-4ab9-a975-7b56309ebcdf sid:5599e05d-8210-4950-89eb-994ae1b33ff4 time:2023-02-13T16:54:29.395Z uid:52d7332f-5dc3-4506-97f1-6be8c761dcb5 user:carson.anderson@goteleport.com events/emitter.go:263
2023-02-13T16:54:29Z INFO             Round trip: GET /, code: 431, duration: 22.336435ms tls:version: 304, tls:resume:false, tls:csuite:1301, tls:server:706c6174666f726d2e74656c65706f72742e7368.teleport.cluster.local forward/fwd.go:182
2023-02-13T16:54:29Z ERRO             Error forwarding to /favicon.ico, err: http2: request header list larger than peer's advertised limit forward/fwd.go:176
```

**Comment by r0mant** [comment]
@rcanderson23 Is this the error log from the proxy or app service?

**Comment by rcanderson23** [comment]
These logs are from app service.

**Comment by pschisa** [comment]
Two users have reported `431 Request Header Fields Too Large` when using application access running on v12.  One user had it previously working on v10.2.0.

**Comment by zmb3** [comment]
Fixed by #23924
